ovs/dal/hybrids/vdisk.py

```diff
@@ -107,33 +108,67 @@ def _dtl_status(self):
         """
         Retrieve the DTL status for a vDisk
         """
-        sd_status = self.info.get('failover_mode', 'UNKNOWN').lower()
+        sd_status = self._info().get('failover_mode', 'UNKNOWN').lower()
         if sd_status == '':
             sd_status = 'unknown'
-        if sd_status != 'ok_standalone':
+        if sd_status not in ['ok_sync', 'ok_standalone']:  # ok_sync or ok_standalone according to voldrv, can still mean incorrect deployment
             return sd_status
 
-        # Verify whether 'ok_standalone' is the correct status for this vDisk
+        # Verify whether 'ok_standalone' or 'ok_sync' is the correct status for this vDisk
         vpool_dtl = self.vpool.configuration['dtl_enabled']
-        if self.has_manual_dtl is True or vpool_dtl is False:
+        if (self.has_manual_dtl is False and vpool_dtl is False) or (self.has_manual_dtl is True and vpool_dtl is True and len(self.domains_dtl_guids) == 0):
             return 'disabled'
 
-        domains = []
-        possible_dtl_targets = set()
-        for sr in StorageRouterList.get_storagerouters():
-            if sr.guid == self.storagerouter_guid:
-                domains = [junction.domain for junction in sr.domains]
-            elif len(sr.storagedrivers) > 0:
-                possible_dtl_targets.add(sr)
+        self.invalidate_dynamics('storagerouter_guid')
```

Comment:
Yes, but isn't that the purpose of the invalidate? Otherwise we can replace all `invalidate_dynamics` logic with `_<property>()` and thats not how its meant to be i guess?